modules/Analytics/components/month-by-month.component.tsx

```diff
@@ -0,0 +1,101 @@
+import dynamic from 'next/dynamic';
+
+// @ts-ignore
+const ResponsiveBarCanvas = dynamic(() => import('@nivo/bar').then(m => m.ResponsiveBarCanvas), { ssr: false });
+
+import { Text, Box as ChackraBox, Tag } from '@chakra-ui/react';
+import { useMemo } from 'react';
+
+import { Box } from '@Components';
+import { IBill } from '@/hooks/useBills';
+import { BillTypes } from '@Modules/Bill/constants/Types';
+
+interface MonthByMonthProps {
+  bills: IBill[];
+}
+
+const keyToText = {
+  rest: 'Retante',
+  income: 'Receita',
+  expense: 'Despesa',
+};
+
+export const MonthByMonth = ({ bills }: MonthByMonthProps) => {
+  const data = useMemo(() => {
+    const billsReduce = bills.reduce((prev, curr) => {
+      const dueDate = new Date(curr.dueDate);
+      const month = `${dueDate.getMonth() + 1}/${dueDate.getFullYear()}`;
```

Comment:
I would rename the variable name `month` to `Formatted Date`, as it doesn't actually represent the month, but rather the `month/year` of the due date